docs/current/sdk/go/959738-get-started.md

```diff
@@ -75,356 +80,440 @@ func build(repoUrl string) error {
-- get a Dagger client with `dagger.Connect()`. This will provide the interface to execute commands against the Dagger engine.
-- clone the git repo. `client.Git()` gives a `GitRepository`, then `.Branch("main").Tree().ID()` will clone the main branch.
-- load the latest golang image with `client.Container().From("golang:latest")`.
-- mount the cloned repo with `.WithMountedDirectory("/src", src)` and set the container's working directory using `.WithWorkdir("/src")`.
-- execute the build command, `go build -o build/` by calling `Container.Exec().ExitCode()`.
-
-1. Try the `test` step of the pipeline by executing the command below from the application directory:
-
-```shell
-go build
-./multibuild https://github.com/kpenfound/greetings-api.git
-```
-
-In the output of this command, you will see Dagger cloning the git repo and running go build on it.
-
-## Step 3: Copy the build output to the host machine
-
-Once the tool has completed the build, it should put that build artifact somewhere to be used. In this step, Dagger will copy the build artifact to the host machine after the build is complete.
-
-1. Update the file `main.go` and some new steps to the `build()` function as shown below. Save the file once done.
-
-```go
-func build(repoUrl string) error {
-  ...
-  // 1. reference to the current working directory on the host
-  workdir := client.Host().Workdir() // <-- New
 
+  // get `golang` image
   golang := client.Container().From("golang:latest")
+
+  // mount cloned repository into `golang` image
   golang = golang.WithMountedDirectory("/src", src).WithWorkdir("/src")
 
-  // 2. Create the output path on the host for the build
-  // -->
+  // build application
   path := "build/"
-  outpath := filepath.Join(".", path)
-  err = os.MkdirAll(outpath, os.ModePerm)
-  if err != nil {
-    return err
-  }
-  // <-- New
-
   golang = golang.Exec(dagger.ContainerExecOpts{
     Args: []string{"go", "build", "-o", path},
   })
 
-  // 3. Get build output from builder
-  // -->
+  // get reference to build output directory in container
   output, err := golang.Directory(path).ID(ctx)
   if err != nil {
     return err
   }
-  // <-- New
 
-  // 4. Write the build output to the host
-  // -->
+  // create build/ directory on host
+  outpath := filepath.Join(".", path)
+  err = os.MkdirAll(outpath, os.ModePerm)
+  if err != nil {
+    return err
+  }
+
+  // get reference to current working directory on the host
+  workdir := client.Host().Workdir()
+
+  // write contents of container build/ directory
+  // to the host working directory
   _, err = workdir.Write(ctx, output, dagger.HostDirectoryWriteOpts{Path: path})
+  // highlight-end
   if err != nil {
     return err
   }
-  // <-- New
 
   return nil
 }
 ```
 
-With this new code, the tool will now write the build artifact to the host after the build is complete.
+The revised `build()` function is the main workhorse here, so let's step through it in detail.
```

Comment:
When I view the actual website I find it hard to jump back and forth between the explanation here and the code.

What do you think about inlining this information into comments in the code itself? I think it will require reducing the verbosity a little bit but it would make it much easier to follow overall IMO.

Reply:
I see what you mean but somehow (maybe this is subjective) I don't think inlining the information *only* in the comments is a good idea. It feels a bit like we are artificially constraining ourselves, since code comments need to be short. Having the explanation in the following text allows us to add more detail/context where needed, as well as important notes, tips etc. However I tried going the reverse approach to what you suggested ie. replicating the relevant lines of code in the text explanation following the code to avoid the jumping back and forth. Interested to get some feedback if this idea makes it better or worse.

Reply:
@sipsma LMK your thoughts on above. I'll wait to hear back before merging this, in case this specific change needs to be reverted or needs another adjustment.

Reply:
I'll go ahead and merge it, we can revert it later if needed.

Reply:
Sorry @vikram-dagger I just caught up on this in my queue, didn't know you were waiting on me. Your approach makes sense though!